test/api/test_assign.cpp

```diff
@@ -0,0 +1,18 @@
+#include <migraphx/migraphx.h>
+#include <migraphx/migraphx.hpp>
+#include "test.hpp"
+
+TEST_CASE(shape_assign)
```

Comment:
Tests for all other types e.g. `Arguments`, would be similar, you can add tests for all / some of them to increase test coverage.